src/Workspaces/Remote/ServiceHub/Services/BrokeredServiceBase.cs

```diff
@@ -94,6 +105,13 @@ protected ValueTask<T> RunServiceAsync<T>(Func<CancellationToken, ValueTask<T>>
             return RunServiceImplAsync(implementation, cancellationToken);
         }
 
+        protected ValueTask<T> RunServiceWithSolutionAsync<T>(
```

Comment:
>RunServiceWithSolutionAsync [](http://example.com/codeflow?start=31&length=27)

I'd call this RunServiceAsync - we'd have two overloads.

Reply:
Done!

Reply:
done.